reference/constraints/Range.rst

```diff
@@ -346,15 +346,17 @@ maxMessage
 **type**: ``string`` **default**: ``This value should be {{ limit }} or less.``
 
 The message that will be shown if the underlying value is more than the
-`max`_ option, and no `min`_ option has been defined (if both are defined, use
-`notInRangeMessage`_).
+`max`_ option, and **no** `min`_ option has been defined.
+
+If **both** are defined, use `notInRangeMessage`_.
 
 You can use the following parameters in this message:
 
 ===============  ==============================================================
 Parameter        Description
 ===============  ==============================================================
-``{{ limit }}``  The upper limit
+``{{ limit }}``  The upper limit (if only `max`_ is defined)
+``{{ max }}``    The upper limit (if both `min`_ and `max`_ are defined)
```

Comment:
This doesn't look right to me. I don't see where `{{ max }}` is actually used.

Reply:
it's used here https://github.com/symfony/validator/blob/d3a98390b77b2a8854c7b5d9058588d4b45794fb/Constraints/RangeValidator.php#L114

Reply:
There it's only used if both `min` and `max` are configured where you shouldn't use the `maxMessage`, but should use the `notInRangeMessage` instead.

Reply:
Yes that's exactly the reason I opened this MR. Because I thought this is missing in documentation. Not knowing that one should not use it anymore. The reason for that is not clear to me though.

Reply:
I would have expected that you would see a deprecation message if you were using both the `min` and `max` option together with one or both of `minMessage` or `maxMessage`. Wasn't that the case for you?

Reply:
Yes, that makes sense. I just checked my project but did not get any deprecation messages - at least when the form is submitted.

Reply:
Would be interesting to see why that's the case. Which exact version do you use? Would you be able to create a small project that allows to reproduce it?

Reply:
Symfony version is 5.2.0. I see what I can do to give you a full example. For now, this is the assertion I am using if this helps in any way:

```
    /**
     * @var mixed
     * @Assert\Sequentially(
     *     @Assert\NotNull(
     *          message="Bitte geben Sie die Anzahl an.",
     *          groups={"invoice_address"}
     *      ),
     *     @Assert\Type(
     *          type="integer",
     *          message="Die Anzahl muss eine ganze Zahl sein.",
     *          groups={"invoice_address"}
     *     ),
     *     @Assert\Range(
     *          min=5,
     *          max=1000,
     *          minMessage="Die Anzahl muss größer als {{ min }} sein.",
     *          maxMessage="Die Anzahl darf max. {{ max }} betragen",
     *          groups={"invoice_address"}
     *     )
     * )
     */
    public $quantity;
    ```

Reply:
I created a sample project here https://github.com/bartrail/symfony-range-deprecation
you can start it up using vagrant and then run the curl command posted in the README.md 
if you open the profiler afterwards, you will see that there is no deprecation message shown.

Reply:
Thanks, that helped to understand what is going on. In fact the deprecation is triggered. But the constraint is only instantiated once and then cached. That's why the deprecation is not displayed in the profiler. I don't know yet how we can fix that. Could you please open an issue in the code repository so that we can keep track of this?